File: 151.java
class Solution {
    public String reverseWords(String s) {

        StringBuilder result = new StringBuilder(s.length() - 1);
        char arr[] = s.toCharArray();

        int fast = s.length() - 1;

        while(fast >= 0 && arr[fast] == ' ') --fast;

        int slow = fast;

        while(fast >= 0){

            while(fast >= 0 && (
                arr[fast] >= 'a' && arr[fast] <= 'z' ||
                arr[fast] >= 'A' && arr[fast] <= 'Z' ||
                arr[fast] >= '0' && arr[fast] <= '9'
            )) --fast;
            
            result.append(s.substring(fast + 1, slow + 1) + " ");

            while(fast >= 0 && s.charAt(fast) == ' ') --fast;

            slow = fast;
        }


        return result.toString().trim();
    }
}

File: 151.cpp
#include"LeetCodeLibs.h"

class Solution {
public:
    string reverseWords(string s) {
        string result;
        
        int fast = s.size() - 1;

        while(fast >= 0 && s[fast] == ' ') --fast;

        int slow = fast;

        while(fast >= 0){

            while(fast >= 0 && (
                s[fast] >= 'a' && s[fast] <= 'z' || 
                s[fast] >= 'A' && s[fast] <= 'Z' || 
                s[fast] >= '0' && s[fast] <='9'
            )) --fast; 

            result += s.substr(fast + 1, slow - fast) + ' ';

            while(fast >= 0 && s[fast] == ' ') --fast;
            slow = fast;
        }

        result.pop_back();

        return result;
    }
};


int main(){
    Solution s;
    cout << s.reverseWords("F0m n G wU");
}